src/python/gem5/resources/resource.py

```diff
@@ -554,6 +555,70 @@ def _get_weights_and_simpoints_from_file(
         return simpoint_list, weight_list
 
 
+class SuiteResource(AbstractResource):
+    def __init__(
+        self,
+        workloads: List[Dict[str, Any]],
+        local_path: Optional[str] = None,
+        resource_version: Optional[str] = None,
+        description: Optional[str] = None,
+        source: Optional[str] = None,
+        **kwargs,
+    ) -> None:
```

Comment:
This is missing a doc-string. The dictionary explaining the `workloads` is necessary.